packages/flutter_tools/lib/src/asset.dart

```diff
@@ -367,18 +385,74 @@ DevFSContent _createAssetManifest(Map<_Asset, List<_Asset>> assetVariants) {
   return new DevFSStringContent(JSON.encode(json));
 }
 
-DevFSContent _createFontManifest(Map<String, dynamic> manifestDescriptor,
-                             bool usesMaterialDesign,
-                             bool includeDefaultFonts) {
+List<Map<String, dynamic>>_parseFonts(
+  Map<String, dynamic> manifestDescriptor,
+  bool usesMaterialDesign,
+  bool includeDefaultFonts,
+  PackageMap packageMap, {
+  String packageName
+}) {
   final List<Map<String, dynamic>> fonts = <Map<String, dynamic>>[];
   if (usesMaterialDesign && includeDefaultFonts) {
     fonts.addAll(_getMaterialFonts(AssetBundle._kFontSetMaterial));
   }
   if (manifestDescriptor != null && manifestDescriptor.containsKey('fonts'))
-    fonts.addAll(manifestDescriptor['fonts']);
-  if (fonts.isEmpty)
-    return null;
-  return new DevFSStringContent(JSON.encode(fonts));
+    if (packageName == null) {
+      fonts.addAll(manifestDescriptor['fonts']);
+    } else {
+      fonts.addAll(_parsePackageFonts(
+        manifestDescriptor['fonts'],
+        packageName,
+        packageMap,
+      ));
+    }
+  return fonts;
```

Comment:
This method no longer returns `null` if the fonts map is empty. Does that work with all callers?

Reply:
I believe so.